projects/packages/connection/src/class-manager.php

```diff
@@ -394,19 +394,19 @@ private function internal_verify_xml_rpc_signature() {
 		}
 
 		$signature_details = array(
-			'token'     => isset( $_GET['token'] ) ? wp_unslash( $_GET['token'] ) : '',
-			'timestamp' => isset( $_GET['timestamp'] ) ? wp_unslash( $_GET['timestamp'] ) : '',
-			'nonce'     => isset( $_GET['nonce'] ) ? wp_unslash( $_GET['nonce'] ) : '',
-			'body_hash' => isset( $_GET['body-hash'] ) ? wp_unslash( $_GET['body-hash'] ) : '',
-			'method'    => isset( $_SERVER['REQUEST_METHOD'] ) ? wp_unslash( $_SERVER['REQUEST_METHOD'] ) : null,
-			'url'       => wp_unslash( ( isset( $_SERVER['HTTP_HOST'] ) ? $_SERVER['HTTP_HOST'] : null ) . ( isset( $_SERVER['REQUEST_URI'] ) ? $_SERVER['REQUEST_URI'] : null ) ), // Temp - will get real signature URL later.
-			'signature' => isset( $_GET['signature'] ) ? wp_unslash( $_GET['signature'] ) : '',
+			'token'     => isset( $_GET['token'] ) ? filter_var( wp_unslash( $_GET['token'] ), FILTER_SANITIZE_STRING ) : '',
+			'timestamp' => isset( $_GET['timestamp'] ) ? filter_var( wp_unslash( $_GET['timestamp'] ), FILTER_SANITIZE_NUMBER_INT ) : '',
+			'nonce'     => isset( $_GET['nonce'] ) ? filter_var( wp_unslash( $_GET['nonce'] ), FILTER_SANITIZE_STRING ) : '',
+			'body_hash' => isset( $_GET['body-hash'] ) ? filter_var( wp_unslash( $_GET['body-hash'] ), FILTER_SANITIZE_STRING ) : '',
+			'method'    => isset( $_SERVER['REQUEST_METHOD'] ) ? filter_var( wp_unslash( $_SERVER['REQUEST_METHOD'] ), FILTER_SANITIZE_STRING ) : null,
+			'url'       => filter_var( wp_unslash( ( isset( $_SERVER['HTTP_HOST'] ) ? $_SERVER['HTTP_HOST'] : null ) . ( isset( $_SERVER['REQUEST_URI'] ) ? $_SERVER['REQUEST_URI'] : null ) ), FILTER_SANITIZE_URL ), // Temp - will get real signature URL later.
```

Comment:
Seems like you may as well just use WordPress's `sanitize_url()` instead of `FILTER_SANITIZE_URL`.